pyinfra/facts/files.py

```diff
@@ -204,91 +205,57 @@ class Socket(File):
     type = "socket"
 
 
-class Sha1File(FactBase):
-    """
-    Returns a SHA1 hash of a file. Works with both sha1sum and sha1. Returns
-    ``None`` if the file doest not exist.
-    """
+class HashFileFactBase(FactBase):
+    _raw_cmd: str
+    _regexes: Tuple[str, str]
+
+    def __init_subclass__(cls, digits: int, cmds: List[str], **kwargs) -> None:
+        super().__init_subclass__(**kwargs)
 
-    _regexes = [
-        r"^([a-zA-Z0-9]{40})\s+%s$",
-        r"^SHA1\s+\(%s\)\s+=\s+([a-zA-Z0-9]{40})$",
-    ]
+        raw_hash_cmds = ["%s {0} 2> /dev/null" % cmd for cmd in cmds]
+        raw_hash_cmd = " || ".join(raw_hash_cmds)
+        cls._raw_cmd = "test -e {0} && ( %s ) || true" % raw_hash_cmd
+
+        assert cls.__name__.endswith("File")
+        hash_name = cls.__name__[:-4].upper()
```

Comment:
This is probably a bit too magic, but I can't see a nicer way aside from passing the value as an extra kwarg. Will merge for now as you mention the tests will catch any issues here, more just a code understanding problem we can address in the future.